server/src/main/java/org/opensearch/transport/TcpTransport.java

```diff
@@ -762,12 +762,24 @@ protected void serverAcceptedChannel(TcpChannel channel) {
     protected abstract void stopInternal();
 
     /**
+     * @deprecated use {@link #inboundMessage(TcpChannel, ProtocolInboundMessage)}
      * Handles inbound message that has been decoded.
      *
      * @param channel the channel the message is from
      * @param message the message
      */
+    @Deprecated(since = "2.14.0", forRemoval = true)
     public void inboundMessage(TcpChannel channel, InboundMessage message) {
+        inboundMessage(channel, message);
```

Comment:
This is recursion, you need need to explicitly cast to `ProtocolInboundMessage`:

```
inboundMessage(channel, (ProtocolInboundMessage) message);
```

Reply:
Ah yes! Fixed